src/Rings/mpoly-local.jl

```diff
@@ -30,26 +30,32 @@ end
 struct MPolyElemLoc{T} <: AbstractAlgebra.RingElem where {T}
   frac::AbstractAlgebra.Generic.Frac
   parent::MPolyRingLoc{T}
-  function MPolyElemLoc(f::MPolyElem{T}, m::Oscar.MPolyIdeal) where {T}
-    R = parent(f)
-    (R != base_ring(m)) && error("Parent rings do not match!")
-    r = new{T}(f//R(1), Localization(R, m))
-    return r
-  end
```

Comment:
Actually, this one *should* be inner, at least if want to ensure the `R == base_ring(m)` invariant.

But it must have the correct signatures; this is actually a bug affecting a lot of Oscar.jl code (and perhaps also our other packages that I just discovered earlier this week while discussing the Modules PR: the above should be `function MPolyElemLoc{T}` not `function MPolyElemLoc` otherwise various code constructs are impossible, and writing generic code gets harder. I should write more on this, but I have to leave for home now; I'll write more later.